packages/api/endpoints/s3credentials.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const router = require('express-promise-router')();
+
+const { lambda } = require('@cumulus/common/aws');
+
+/**
+ * Use NGAP's time-based, temporary credential dispensing lambda.
+ *
+ * @param {string} username - earthdata login username
+ * @returns {Promise<Object>} Payload containing AWS STS credential object valid for 1
+ *                   hour.  The credential object contains keys: AccessKeyId,
+ *                   SecretAccessKey, SessionToken, Expiration and can be use
+ *                   for same-region s3 direct access.
+ */
+async function requestTemporaryCredentialsFromNgap(username) {
+  const FunctionName = 'gsfc-ngap-sh-s3-sts-get-keys';
+  const Payload = JSON.stringify({
+    accesstype: 'sameregion',
+    duration: '3600', // one hour max allowed by AWS.
+    rolesession: 'SAME_REGION_ACCESS',
+    userid: username
+  });
+
+  return lambda().invoke({
+    FunctionName,
+    Payload
+  }).promise();
+}
+
+/**
+ * Dispenses time based temporary credentials for same-region direct s3 access.
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Object} the express response object with object containing
+ *                   tempoary s3 credentials for direct same-region s3 access.
+ */
+async function s3credentials(req, res) {
+  const username = req.authorizedMetadata.userName;
+  const credentials = await requestTemporaryCredentialsFromNgap(username);
+  return res.send(credentials.Payload);
```

Comment:
The one thing here is when I tested the response came back as a JSON string and that is inconsistent with the rest of our API which comes back as JSON. I think maybe we should be sending back an object.